src/DocumentDB.ChangeFeedProcessor/Bootstrapping/PartitionSynchronizer.cs

```diff
@@ -142,7 +144,7 @@ private async Task<bool> HasCollectionProvisionedThroughput()
             }
             while (!string.IsNullOrEmpty(response.ResponseContinuation));
 
-            this.hasCollectionProvisionedThroughput = offers.Any(x => x.ResourceId == this.collectionSelfLink);
+            this.hasCollectionProvisionedThroughput = offers.Any(x => x.ResourceLink == this.collectionSelfLink);
```

Comment:
This is actually wrong. collectionSelfLink would be in name format. Offer.ResourceLink would be in RID format.